pkg/aws/s3/objects.go

```diff
@@ -192,3 +193,22 @@ func ListObjectsWithPrefix(s3Session *s3.S3, r *ListObjectsReq) ([]*s3.Object, e
 
 	return res.Contents, nil
 }
+
+func GetObjectPresignedURL(s3Session *s3.S3, r *GetObjectRequest, duration time.Duration) (string, error) {
```

Comment:
Instead of taking duration argument separately, we can make a new request struct for this request.